cmd/protoc-gen-openapi/examples/tests/oneofoptions/openapi.yaml

```diff
@@ -0,0 +1,77 @@
+# Generated with protoc-gen-openapi
+# https://github.com/google/gnostic/tree/master/cmd/protoc-gen-openapi
+
+openapi: 3.0.3
+info:
+    title: Messaging API
+    description: Messaging service
+    version: 0.0.1
+paths:
+    /v1/messages:
+        post:
+            tags:
+                - Messaging
+            operationId: Messaging_CreateMessage
+            requestBody:
+                content:
+                    application/json:
+                        schema:
+                            $ref: '#/components/schemas/Message'
+                required: true
+            responses:
+                "200":
+                    description: OK
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/Message'
+                default:
+                    description: Default error response
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/Status'
+components:
+    schemas:
+        GoogleProtobufAny:
+            type: object
+            properties:
+                '@type':
+                    type: string
+                    description: The type of the serialized message.
+            additionalProperties: true
+            description: Contains an arbitrary serialized message along with a @type that describes the type of the serialized message.
+        Message:
+            required:
+                - plain_text
+            type: object
+            properties:
+                message_id:
+                    type: string
+                body:
+                    $ref: '#/components/schemas/Message_Body'
+                plain_text:
+                    type: string
```

Comment:
Just clarifying: this change flattens the proto oneof into distinct fields for each possibility... or (looking further)